crates/sui-indexer-alt-framework/src/pipeline/concurrent/collector.rs

```diff
@@ -184,7 +184,12 @@ pub(super) fn collector<H: Handler + 'static>(
                 Some(mut indexed) = rx.recv(), if pending_rows < H::MAX_PENDING_ROWS => {
                     // Clear the values of outdated checkpoints, so that we don't commit data to the
                     // store, but can still advance watermarks.
-                    if indexed.checkpoint() < main_reader_lo.wait().await.load(Ordering::Relaxed) {
+                    let reader_lo = main_reader_lo.wait().await.load(Ordering::Relaxed);
+                    metrics
```

Comment:
nit: move this next to the other metric updates below? It being separate makes it seem as if the ordering is relevant when I don't think it is/should be.